torch/functional.py

```diff
@@ -234,7 +257,13 @@ def argmax(input, dim=None, keepdim=False):
         tensor([ 0,  2,  0,  1])
     """
     if dim is None:
-        return torch._argmax(input.contiguous().view(-1), dim=0, keepdim=False)
+        # Instead of squeeze to 1 dimension, do it multiple passes for better perf.
```

Comment:
I'm not 100% convinced that having multiple passes is a clear win. There are different considerations we have to take into account here:
1. You did this, because it exposes more paralellism on the GPU, but this is only because of how our kernel is written. This will likely be slower on the CPU.
2. If the input is actually very small (think of cases when the kernel launch overhead is larger than the actual cost of the whole compute), then this will make the op multiple times slower (because we'll have to launch multiple kernels).